pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/StoolUploadRoute.java

```diff
@@ -0,0 +1,180 @@
+import org.broadinstitute.dsm.exception.FileWrongSeparator;
+import org.broadinstitute.dsm.exception.UploadLineException;
+import org.broadinstitute.dsm.model.StoolUploadObject;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.statics.UserErrorMessages;
+import org.broadinstitute.dsm.util.SystemUtil;
+import org.broadinstitute.dsm.util.UserUtil;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+
+public class StoolUploadRoute extends RequestHandler {
+    private static final Logger logger = LoggerFactory.getLogger(StoolUploadRoute.class);
+
+
+    private static final String PARTICIPANT_ID = "participantId";
+    private static final String MF_BARCODE = "mfBarcode";
+    private static final String RECEIVE_DATE = "receiveDate";
+
+    @Override
+    protected Object processRequest(Request request, Response response, String userId) throws Exception {
+        QueryParamsMap queryParams = request.queryMap();
+        String realm;
+
+        if (queryParams.value(RoutePath.REALM) != null) {
+            realm = queryParams.get(RoutePath.REALM).value();
+        } else {
+            throw new RuntimeException("No realm query param was sent");
+        }
+        String userIdRequest = UserUtil.getUserId(request);
+
+        if (UserUtil.checkUserAccess(realm, userId, "kit_upload", userIdRequest)) {
+            String content = request.body();
+
+
+            try {
+                List<StoolUploadObject> stoolUploadContent;
+
+                try {
+                    stoolUploadContent = isFileValid(content);
+                } catch (Exception e) {
+                    response.status(500);
+                    return e.getMessage();
+                }
+
+                final List<StoolUploadObject> stoolUploadObjects = stoolUploadContent;
+                if (stoolUploadObjects.isEmpty()) {
+                    return "Text file was empty or couldn't be parsed to the agreed format";
+                }
+
+                logger.info("Text file was uploaded and parsed successfully");
+
+                StoolUploadDao stoolUploadDao = new StoolUploadDao();
+                stoolUploadObjects.forEach(stoolUploadObject -> {
+
+                    String participantId = stoolUploadObject.getParticipantId();
+                    String mfBarcode = stoolUploadObject.getMfBarcode();
+                    String receiveDate = stoolUploadObject.getReceiveDate();
+
+                    Optional<StoolUploadDto> stoolUploadDto = stoolUploadDao.getStoolUploadDto(participantId, mfBarcode);
+
+                    if (stoolUploadDto.isPresent()) {
+                        logger.info("Successfully generated kits, trying to update the table with provided parameters...");
+                        stoolUploadDao.updateKitData(receiveDate, stoolUploadDto.get().getKitId(), mfBarcode);
```

Comment:
I think I would like to change that to having the select in the update. 
I worry that if they are uploading a file with 100 kits we need to call the db 100*2 
If you combine the update and select query we only need to do 1 call per kit.

Reply:
agree here, instead of doing DB transaction for each item of the list, introduce bulk upload in Dao class, which will insert list of samples at once.